### PR TITLE
Improve resty package

### LINKED ISSUE #737: improve `resty` package
as per @peterlimg mentioned on https://github.com/0chain/gosdk/pull/718#discussion_r1089708101

> We'd better use the ctx instead of coming up with a new timeout, the API caller should be able to control the timeout with the ctx param. Anyway, that will require the refactoring of the. resty package to drop the timeout param or make it optional, which could be done in another thread.

### PR DESCRIPTION
**Pull request by aniketsingh03**
### Changes
This PR makes the following enhancements/fixes following bugs in the resty package:
- It removes the custom timeout param that was being sent while creating resty object. The timeout will now be controlled by the API caller/client through the request context. However, a default timeout of 10s (DefaultRequestTimeout) is still there for requests which do not have any timeout set.
- Fixes a bug in the code where errors returned by the interceptor weren't getting returned to `resty.Wait()`
- Adds/modifies unit tests for resty's functionality

### Fixes
Fixes #737 

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by aniketsingh03** [comment]
Can you guys review this @dabasov @cnlangzi @peterlimg?

File: zcncore/transaction_query_base.go [hunk 1]
```diff
@@ -119,7 +119,12 @@ func (tq *TransactionQuery) checkHealth(ctx context.Context, host string) error
 	}
 
 	// check health
-	r := resty.New(resty.WithTimeout(5 * time.Second))
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	go func() {
+		<-ctx.Done()
+		cancel()
+	}()
```
Comment (peterlimg) [inline zcncore/transaction_query_base.go]: We don't need this goroutine, call `defer cancel()` would be fine.

**Review by peterlimg — DISMISSED** [review]

File: zcncore/transaction_query_base.go (hunk 1 above)
Comment (aniketsingh03) [reply zcncore/transaction_query_base.go]: Done

**Review by dabasov — COMMENTED** [review]
lgtm

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | improve-resty-package  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | true  |
| system-tests  | master  |
| gosdk  | improve-resty-package  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by aniketsingh03** [comment]
Thanks!

**Pull request merged**